addons/survey/models/survey.py

```diff
@@ -0,0 +1,1159 @@
+    _description = 'Survey Stage'
+    _order = 'sequence,id'
+
+    name = fields.Char(required=True, translate=True)
+    sequence = fields.Integer(default=1)
+    closed = fields.Boolean(default=False, help="If closed, people won't be able to answer to surveys in this column.")
+    fold = fields.Boolean(string="Folded in kanban view")
+
+    _sql_constraints = [
+        ('positive_sequence', 'CHECK(sequence >= 0)', 'Sequence number MUST be a natural')
+    ]
+
+
+class Survey(models.Model):
+    '''Settings for a multi-page/multi-question survey.
+    Each survey can have one or more attached pages, and each page can display
+    one or more questions.
+    '''
+
+    _name = 'survey.survey'
+    _description = 'Survey'
+    _rec_name = 'title'
+    _inherit = ['mail.thread', 'ir.needaction_mixin']
+
+    def _default_stage(self):
+        return self.env['survey.stage'].search([], limit=1)
+
+    # Model fields #
+
+    title = fields.Char(required=True, translate=True)
+    page_ids = fields.One2many('survey.page', 'survey_id', string='Pages', copy=True)
+    stage_id = fields.Many2one('survey.stage', string="Stage", copy=False,
+        default='_default_stage')
+    auth_required = fields.Boolean(string='Login required',
+        help="Users with a public link will be requested to login before taking part to the survey",
+        oldname="authenticate")
+    users_can_go_back = fields.Boolean(help="If checked, users can go back to previous pages.")
+    tot_sent_survey = fields.Integer(compute="_compute_tot_sent_survey",
+        string="Number of sent surveys")
+    tot_start_survey = fields.Integer(compute="_compute_tot_start_survey",
+        string="Number of started surveys")
+    tot_comp_survey = fields.Integer(compute="_compute_tot_comp_survey",
+        string="Number of completed surveys")
+    description = fields.Html(translate=True,
+        help="A long description of the purpose of the survey")
+    color = fields.Integer(string='Color Index')
+    user_input_ids = fields.One2many('survey.user_input', 'survey_id',
+        string='User responses', readonly=True)
+    designed = fields.Boolean(compute='_compute_designed', string="Is designed?")
+    public_url = fields.Char(compute='_compute_public_url',
+        string="Public link")
+    public_url_html = fields.Char(compute='_compute_public_url_html',
+        string="Public link (html version)")
+    print_url = fields.Char(compute='_compute_print_url',
+        string="Print link")
+    result_url = fields.Char(compute='_compute_result_url',
+        string="Results link")
+    email_template_id = fields.Many2one('mail.template',
+        string='Email Template')
+    thank_you_message = fields.Html(translate=True,
+        help="This message will be displayed when survey is completed")
+    quizz_mode = fields.Boolean(string='Quiz mode')
+
+    ## Function fields ##
+
+    @api.multi
+    def _compute_designed(self):
+        for survey in self:
```

Comment:
noooooooooooooooooooooooooooooooooooooooo

keep sql queries for practical performance issues....